test/uniqBy.js

```diff
@@ -0,0 +1,30 @@
+var assert = require('assert');
+
+var R = require('..');
+
+
+describe('uniqBy', function() {
+
+  var objs = [
+    {i: 1, v: 1}, {i: 2, v: 2}, {i: 3, v: 3}, {i: 4, v: 4}, {i: 5, v: 5},
+    {i: 6, v: 6}, {i: 7, v: 7}, {i: 8, v: 8}, {i: 9, v: 9}, {i: 10, v: 10}
+  ];
+
+  var objs2 = [
+    {i: 1, v: 1}, {i: 2, v: 2}, {i: 2, v: 3}, {i: 4, v: 4}, {i: 5, v: 5},
+    {i: 6, v: 6}, {i: 6, v: 7}, {i: 6, v: 8}, {i: 9, v: 9}, {i: 5, v: 10}
+  ];
+
+  it('returns a set from any array (i.e. purges duplicate elements) based on predicate', function() {
+    assert.deepEqual(R.uniqBy(R.prop('i'), objs), objs);
+    assert.deepEqual(R.uniqBy(R.prop('i'), objs2), [{i: 1, v: 1}, {i: 2, v: 2}, {i: 4, v: 4}, {i: 5, v: 5}, {i: 6, v: 6}, {i: 9, v: 9}]);
+  });
+
+  it('keeps elements from the left', function() {
+    assert.deepEqual(R.uniqBy(R.prop('i'), [{i: 1}, {i: 2}, {i: 3}, {i: 4}, {i: 1}]), [{i: 1}, {i: 2}, {i: 3}, {i: 4}]);
```

Comment:
I agree with all the notes from @davidchambers except this one.  This assertion doesn't really match the code, but the assertion is a good one to demonstrate, and something like this might be better:

```js
R.uniqBy(R.prop('i'), [{i: 1, j: 10}, {i: 2, j: 20}, {i: 3, j: 30}, {i: 1, j: 50}, {i: 2, j: 2}]);
//=> [{i: 1, j: 10}, {i: 2, j: 20}, {i: 3, j: 30}]
```